src/main/java/dev/latvian/mods/kubejs/util/MobEffectUtil.java

```diff
@@ -0,0 +1,46 @@
+package dev.latvian.mods.kubejs.util;
+
+import dev.latvian.mods.kubejs.typings.Info;
+import net.minecraft.core.Holder;
+import net.minecraft.world.effect.MobEffect;
+import net.minecraft.world.effect.MobEffectInstance;
+import org.jetbrains.annotations.Contract;
+import org.jetbrains.annotations.NotNull;
+
+public class MobEffectUtil {
+	@Contract("_ -> new")
+	@Info("Creates a new instance from the old one")
+	public static @NotNull MobEffectInstance of(MobEffectInstance oldInstance) {
+		return new MobEffectInstance(oldInstance)
+	}
+
+    @Contract("_ -> new")
+	@Info("Creates an instance for the given effect")
+    public static @NotNull MobEffectInstance of(Holder<MobEffect> effect) {
+        return new MobEffectInstance(effect);
+    }
+
+    @Contract("_, _ -> new")
+	@Info("Creates an instance for the given effect and duration")
+    public static @NotNull MobEffectInstance of(Holder<MobEffect> effect, int duration) {
+        return new MobEffectInstance(effect, duration);
+    }
+
+    @Contract("_, _, _ -> new")
+	@Info("Creates an instance for the given effect, duration and amplifier")
```

Comment:
These amplifier ones should specify where the base point is (ie for the effect command amplifier 0 is a level 1 effect while 1 is actually level 2)